### ISSUE TITLE
Additional fraud meta box status handling for fraud outcomes

### ISSUE THREAD

**Issue by jessepearson** [Automattic/woocommerce-payments#5878]
### Description
This is meant to be a follow up to https://github.com/Automattic/woocommerce-payments/pull/5782
There needs to be fraud meta box  status handling for when authorizations are cancelled or expire. 

### Acceptance criteria

Need status handling for:
* Auth/capture enabled and:
  * Held for review, auth cancelled.
  * Held for review, auth expired.
  * Allowed, auth cancelled.
  * Allowed, auth expired.
* Auth/capture disabled and:
  * Held for review, auth cancelled.
  * Held for review, auth expired.

There may be overlap with these, but we need meta box statuses and messages for when these instances occur. This would be so the meta box matches the status of the order and isn't telling the merchant they need to capture a charge when they no longer can.

**Comment by jessepearson** [comment]
Created the initial code for the updated statuses. The ones added differ from the ones in the initial notes due to there was likely too much redundancy with them. A couple tests were failing, so the tests will need to be checked and updated, along with new ones added.

- [x] Add new statuses and messages
- [x] Add new tests
- [x] Update failing tests

**Comment by jessepearson** [comment]
There are changes in https://github.com/Automattic/woocommerce-payments/pull/5900 that need to be updated in this PR once it is merged. 

I also came across the issue where UPE methods created a bug that had to be fixed on the server: p1680111123941939-slack-C01B8KNUYSW . I found this when I started writing manual testing for this issue.

UPE methods also do not set the payment method as `woocommerce_payments` in the order, and that's something we look for in the meta data. They are set like `woocommerce_payments_bancontact`, this means additional changes are needed in order to make sure we're displaying the proper notice in the meta box.

- [x] Include #5900 once merged.
  - [x] Update what's needed with the code once available.
- [x] Update meta box to have a notice for when the payment method is not a card.
- [x] Update test coverage.
- [x] Write manual testing steps.

**Comment by jessepearson** [comment]
I was in the process of writing unit and automated tests when I found a flaw in how we were determining what message to display in the order. It seems that legacy UPE and split UPE add data differently to the order meta, leaving us not being able to accurately determine the payment method, and that doesn't allow us to display the `not_card` message.

I have created a new PR to include this data in the intent object, and once that's merged, I should be able to finish this up.
#5940

**Comment by jessepearson** [comment]
Yesterday I was working through making sure everything was working in the code and found that legacy UPE bypasses "review" filtering. I have created a new issue for this here: https://github.com/Automattic/woocommerce-payments-server/issues/3219

Without this functionality being present in the client, I will not be able to bring my PR to full completion. I will get it to the point to where everything works, excluding that functionality, and will make a note about the issue in my PR.

The largest hurdle here is orders can be in many states, and there are currently 4 payment paths that can be taken, and each of those needs to properly set different statuses on the order to display relevant data. The 4 paths are:
* non-UPE
* legacy UPE
* split UPE
* in person payments

Webhooks can also update an order's status to completed, and sometimes webhooks arrive before the processing completes on the website itself. This means webhook actions must match what everything else does.

**Comment by jessepearson** [comment]
I've updated most of the manual testing that is needed for the PR, but there's still more to go, I only got through the non-UPE testing, and I haven't covered how to test expired auths.

- [x] Finish writing manual testing instructions for all possible configurations or store states.
- [x] Finish up unit tests.

These two list items coincide with ones above.

**Comment by jessepearson** [comment]
Nearly all manual testing has been added to the PR now, however, I found another possible bug and am waiting for confirmation from the team to see if it actually is a bug.
p1680708928940659-slack-C01B8KNUYSW

**Comment by jessepearson** [comment]
Manual testing steps have been added.
Unit tests have been updated, and everything is passing.
PR has been set to review status.
